extensions/wikia/AbPerfTesting/classes/Hooks.class.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+namespace Wikia\AbPerfTesting;
+
+use \Wikia\Logger\WikiaLogger;
+
+class Hooks {
+
+	/**
+	 * Initialize the experiment and set all required tracking things
+	 *
+	 * @param string $experimentName
+	 * @param array $experimentConfig
+	 */
+	private static function startExperiment( $experimentName, Array $experimentConfig ) {
+		wfDebug( sprintf( "%s[%s] using %s class with %s params\n",
+			__METHOD__, $experimentName, $experimentConfig['handler'], json_encode( $experimentConfig['params'] ) ) );
+
+		new $experimentConfig['handler']( $experimentConfig['params'] ? : [] );
+
+		// mark a transaction with an experiment name
+		\Transaction::getInstance()->set( \Transaction::PARAM_AB_PERFORMANCE_TEST, $experimentName );
+
+		// mark a transaction using UA's custom dimensions
+		global $wgHooks;
+		$wgHooks['WikiaSkinTopScripts'][] = function( Array &$vars, &$scripts ) use ( $experimentName ) {
+			$val = \Xml::encodeJsVar( $experimentName );
+			$scripts .= \Html::inlineScript( "_gaq.push(['set', 'dimension20', {$val}]);" );
+			return true;
+		} ;
+
+		/*
+		 * Start the session to bypass CDN cache
+		 *
+		 * We don't want to polute the CDN cache with the A/B performance testing tracking data.
+		 * As the test are run for only a small subset of the traffic, start the session for client
+		 * that are in the test groups to bypass the CDN cache.
+		 */
+		if ( session_id() == '' ) {
+			wfSetupSession();
+			wfDebug( __METHOD__ . " - session started\n" );
+
+			// log started sessions
+			global $wgUser;
+			WikiaLogger::instance()->info( __METHOD__, [
+				'experiment' => $experimentName,
+				'session_id' => session_id(),
+				'is_anon' => $wgUser->isAnon()
+			] );
+		}
```

Comment:
I think the results may then be opinionated towards heavy readers behavior and don't take casual readers into account fully. My best guess would be that casual readers may hit more popular pages where the hit ratio can be much bigger than 10%.